### PR TITLE
Add Config::get_profile()

### PR DESCRIPTION
**Pull request by nabijaczleweli**
Might be required for, e.g., getting the right artifact path

### CODE REVIEW

**Comment by alexcrichton** [comment]
Thanks for this! I wonder if it might make more sense on `Config` though as a standalone method? That way it can take into account any configuration done on `Config`

**Comment by nabijaczleweli** [comment]
Done

File: src/lib.rs [hunk 1]
```diff
@@ -100,6 +100,73 @@ pub fn build<P: AsRef<Path>>(path: P) -> PathBuf {
...
 
 impl Config {
+    /// Infer `CMAKE_BUILD_TYPE` from Rust's compilation profile
+    ///
+    /// * if `opt-level=0` then `CMAKE_BUILD_TYPE=Debug`,
+    /// * if `opt-level={1,2,3}` and:
+    ///   * `debug=false` then `CMAKE_BUILD_TYPE=Release`
+    ///   * otherwise `CMAKE_BUILD_TYPE=RelWithDebInfo`
+    /// * if `opt-level={s,z}` then `CMAKE_BUILD_TYPE=MinSizeRel`
+    pub fn default_build_profile(&self) -> &'static str {
```
Comment (alexcrichton) [inline src/lib.rs:110]: Could this perhaps be `get_profile` and respect the value in `self.profile` if it's already configured?

**Comment by nabijaczleweli** [comment]
Done

File: src/lib.rs [hunk 2]
```diff
@@ -100,6 +101,78 @@ pub fn build<P: AsRef<Path>>(path: P) -> PathBuf {
...
 
 impl Config {
+    /// Return explicitly set profile or infer `CMAKE_BUILD_TYPE` from Rust's compilation profile.
+    ///
+    /// * if `opt-level=0` then `CMAKE_BUILD_TYPE=Debug`,
+    /// * if `opt-level={1,2,3}` and:
+    ///   * `debug=false` then `CMAKE_BUILD_TYPE=Release`
+    ///   * otherwise `CMAKE_BUILD_TYPE=RelWithDebInfo`
+    /// * if `opt-level={s,z}` then `CMAKE_BUILD_TYPE=MinSizeRel`
+    pub fn get_profile(&self) -> Cow<str> {
```
Comment (alexcrichton) [inline src/lib.rs:111]: I think this could return `&str` perhaps? I don't think that there's anywhere that necessarily needs to return a `String`

Comment (nabijaczleweli) [reply src/lib.rs]: You're right, of course. Done.

**Comment by alexcrichton** [comment]
Thanks!

**Pull request merged**